File: OddWorksDeviceSDK/src/main/java/io/oddworks/device/request/OddCallback.java
package io.oddworks.device.request;

/**
 * Created by brkattk on 9/15/15.
 */

/**
 * @param <T> type of entity returned on success
 */
public interface OddCallback<T> {
    public void onSuccess(T entity);
    public void onFailure(Exception exception);
}

File: device-sdk/src/main/java/io/oddworks/device/request/OddCallback.kt
package io.oddworks.device.request


/**
 * Specify the type of OddResource to return on success.
 *
 * <p>When fetching a set of potentially mixed resources,
 * use {@code LinkedHashSet<OddResource>}
 *
 * @param  type of {@link OddResource} returned on success
 */
interface OddCallback<T> {
    fun onSuccess(resource: T)
    fun onFailure(exception: Exception)
}